release/release.py

```diff
@@ -26,9 +26,9 @@ def get_version() -> str:
     with pathlib.Path(release_folder / "pyproject.toml").open(mode="rb") as py_project_toml_fp:
         py_project_toml = tomllib.load(py_project_toml_fp)
     version_re = py_project_toml["tool"]["bumpversion"]["parse"]
-    version_tags = [tag for tag in repo.tags if re.match(version_re, tag.tag.tag.strip("v"), re.MULTILINE)]
+    version_tags = [tag for tag in repo.tags if tag.tag and re.match(version_re, tag.tag.tag.strip("v"), re.MULTILINE)]
     latest_tag = sorted(version_tags, key=lambda tag: tag.commit.committed_datetime)[-1]
-    return latest_tag.tag.tag.strip("v")
+    return latest_tag.tag.tag.strip("v") if latest_tag.tag else "?"
```

Comment:
But what should we do with this, is it useful?

Reply:
Since `latest_tag.tag` can be `None` (but no idea in what circumstances) I had to do it like this to keep `mypy` happy. Maybe it's better to raise an exception, I'll do that.

Reply:
Oh wait, mypy thinks `latest_tag.tag` can be `None`, but mypy is wrong, so we can just `cast`:

```python
    version_tags = [tag for tag in repo.tags if tag.tag and re.match(version_re, tag.tag.tag.strip("v"), re.MULTILINE)]
    latest_tag = sorted(version_tags, key=lambda tag: tag.commit.committed_datetime)[-1]
    # We cast latest_tag.tag to TagObject because we know it cannot be None, given how version_tags is constructed
    return cast(git.TagObject, latest_tag.tag).tag.strip("v")
```

Reply:
It can indeed be none if you have a tag locally which hasn't been pushed to upstream. I encountered this when testing the changes to the release script :-) But I think it's fine if we cast, it's easy enough to deal with the exception